openhands/server/session/session.py

```diff
@@ -135,9 +135,26 @@ async def initialize_agent(
         self.config.search_api_key = settings.search_api_key
 
         # NOTE: this need to happen AFTER the config is updated with the search_api_key
-        self.config.mcp = settings.mcp_config or MCPConfig(
-            sse_servers=[], stdio_servers=[]
+        # Handle MCP config priority: self.config.mcp > settings.mcp_config > empty MCPConfig
+        config_has_mcp = (
+            self.config.mcp.sse_servers
+            or self.config.mcp.stdio_servers
+            or self.config.mcp.shttp_servers
         )
+        settings_has_mcp = settings.mcp_config is not None
+
+        if config_has_mcp and settings_has_mcp:
+            self.logger.warning(
+                'Both config.mcp and settings.mcp_config are present. '
+                'Using config.mcp and ignoring settings.mcp_config.'
+            )
+            # Keep self.config.mcp as is
+        elif settings_has_mcp and settings.mcp_config is not None:
```

Comment:
```suggestion
        elif settings_has_mcp:
```
Nit: this was the same with the value of `settings_has_mcp`, if I see this right?